src/sm9/SM9DSA.c

```
@@ -1,184 +1,180 @@
+{
+	BNField12 g,w;
+	CBigInt r,l;
+	unsigned int len1,len2;
+	BYTE *msg;
+	Pairing_opt(&g,P_pub,P1);
+	Get(&r,"033C8616B06704813203DFD00965022ED15975C662337AED648835DC4B1CBE",HEX);  // r = rand()
+	F12_exp(&w,g,r);
+	len1 = strlen((const char*)M);
+	len2 = len1 + 384;  // A 12th extension field element requires 32*12 = 384 bytes
+	msg = (BYTE*)malloc(len2);
+	memcpy(msg,M,len1);
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)